Mathlib/Geometry/Manifold/VectorField.lean

```diff
@@ -0,0 +1,317 @@
+/-
+Copyright (c) 2024 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.Calculus.VectorField
+import Mathlib.Geometry.Manifold.ContMDiffMFDeriv
+import Mathlib.Geometry.Manifold.MFDeriv.NormedSpace
+import Mathlib.Geometry.Manifold.VectorBundle.MDifferentiable
+
+/-!
+# Vector fields in manifolds
+
+We study functions of the form `V : Π (x : M) → TangentSpace I x` on a manifold, i.e.,
+vector fields.
+
+We define the pullback of a vector field under a map, as
+`VectorField.mpullback I I' f V x := (mfderiv I I' f x).inverse (V (f x))`
+(together with the same notion within a set).
+
+We start developing API around this notion.
+
+All these are given in the `VectorField` namespace because pullbacks, Lie brackets, and so on,
+are notions that make sense in a variety of contexts. We also prefix the notions with `m` to
+distinguish the manifold notions from the vector spaces notions.
+
+For notions that come naturally in other namespaces for dot notation, we specify `vectorField` in
+the name to lift ambiguities. For instance, the fact that the Lie bracket of two smooth vector
+fields is smooth will be `ContMDiffAt.mlieBracket_vectorField`.
+
+Note that a smoothness assumption for a vector field is written by seeing the vector field as
+a function from `M` to its tangent bundle through a coercion, as in:
+`MDifferentiableWithinAt I I.tangent (fun y ↦ (V y : TangentBundle I M)) s x`.
+-/
+
+open Set Function Filter
+open scoped Topology Manifold ContDiff
+
+noncomputable section
+
+/- We work in the `VectorField` namespace because pullbacks, Lie brackets, and so on, are notions
+that make sense in a variety of contexts. We also prefix the notions with `m` to distinguish the
+manifold notions from the vector spaces notions. For instance, the Lie bracket of two vector
+fields in a manifold is denoted with `VectorField.mlieBracket I V W x`, where `I` is the relevant
+model with corners, `V W : Π (x : M), TangentSpace I x` are the vector fields, and `x : M` is
+the basepoint.
+-/
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {H : Type*} [TopologicalSpace H] {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {I : ModelWithCorners 𝕜 E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
+  {H' : Type*} [TopologicalSpace H'] {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  {I' : ModelWithCorners 𝕜 E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M']
+  {H'' : Type*} [TopologicalSpace H''] {E'' : Type*} [NormedAddCommGroup E''] [NormedSpace 𝕜 E'']
+  {I'' : ModelWithCorners 𝕜 E'' H''}
+  {M'' : Type*} [TopologicalSpace M''] [ChartedSpace H'' M'']
+  {f : M → M'} {s t : Set M} {x x₀ : M}
+
+instance {n : ℕ} [n.AtLeastTwo] [IsManifold I (minSmoothness 𝕜 (ofNat(n))) M] :
```

Comment:
Should these instances go in another file?
Should they be scoped (to e.g. the `Manifold` or `VectorField` namespace)?